modules/veteran/app/models/veteran/service/representative.rb

```diff
@@ -112,27 +112,17 @@ def diff(rep_data)
         end
       end
 
-      def self.get_suffixes(first_name, last_name)
-        first_suffix = first_name.split.last if first_name.split.count > 1
-        last_suffix = last_name.split.last if last_name.split.count > 1
-        [first_suffix, last_suffix]
-      end
-
-      def self.get_representatives(first_name, last_name)
-        suffixes = get_suffixes(first_name, last_name)
-
-        representatives = where('lower(first_name) = ? AND lower(last_name) = ?', first_name&.downcase,
-                                last_name&.downcase)
-
-        if representatives.blank? && suffixes.any?
-          # check without suffix
-          first_name = first_name.delete(suffixes[0]).strip if suffixes[0].present?
-          last_name = last_name.delete(suffixes[1]).strip if suffixes[1].present?
+      def self.get_representatives(first_name, last_name, suffix)
+        representatives = if suffix.present?
+                            last_and_suffix = "#{last_name} #{suffix.downcase}"
 
-          representatives = where('lower(first_name) = ? AND lower(last_name) = ?', first_name&.downcase,
-                                  last_name&.downcase)
+                            where('lower(first_name) = ? AND lower(last_name) = ?', first_name&.downcase,
+                                  last_and_suffix&.downcase)
+                          end
+        if representatives.blank?
+          where('lower(first_name) = ? AND lower(last_name) = ?', first_name&.downcase,
+                last_name&.downcase)
```

Comment:
We aren't returning anything if we find representatives with a suffix (i.e., the function will return nil if `representatives` is not blank). Should we add a return statement either after line 121 or 124?